rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -0,0 +1,727 @@
+import org.ethereum.core.Block;
+import org.ethereum.core.Repository;
+import org.ethereum.util.RLP;
+import org.ethereum.util.RLPList;
+import org.ethereum.vm.LogInfo;
+import org.ethereum.vm.PrecompiledContracts;
+
+import co.rsk.bitcoinj.core.*;
+import co.rsk.bitcoinj.script.Script;
+import co.rsk.bitcoinj.script.ScriptChunk;
+import co.rsk.config.BridgeRegTestConstants;
+import co.rsk.crypto.Keccak256;
+import co.rsk.peg.btcLockSender.BtcLockSenderProvider;
+import co.rsk.peg.pegininstructions.PeginInstructionsProvider;
+import co.rsk.core.RskAddress;
+import co.rsk.peg.utils.*;
+import co.rsk.test.builders.BridgeSupportBuilder;
+
+import com.google.common.collect.Lists;
+import org.bouncycastle.util.encoders.Hex;
+import org.bouncycastle.asn1.x9.X9ECParameters;
+import org.bouncycastle.crypto.ec.CustomNamedCurves;
+import org.bouncycastle.crypto.params.ECDomainParameters;
+import org.bouncycastle.crypto.params.ECPrivateKeyParameters;
+import org.bouncycastle.crypto.signers.ECDSASigner;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static co.rsk.peg.PegTestUtils.*;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+public class BridgeSupportAddSignatureTest extends BridgeSupportTestBase {
+
+    protected ActivationConfig.ForBlock activationsBeforeForks;
+    protected ActivationConfig.ForBlock activationsAfterForks;
+    private static final RskAddress contractAddress = PrecompiledContracts.BRIDGE_ADDR;
+
+    @Before
+    public void setUpOnEachTest() {
+        activationsBeforeForks = ActivationConfigsForTest.genesis().forBlock(0);
+        activationsAfterForks = ActivationConfigsForTest.all().forBlock(0);
+        bridgeSupportBuilder = new BridgeSupportBuilder();
+    }
+
+    @Test
+    public void addSignature_fedPubKey_belongs_to_active_federation() throws Exception {
+        //Setup
+        FederationSupport mockFederationSupport = mock(FederationSupport.class);
+
+        // Creates new federation
+        List<BtcECKey> federation1Keys = Arrays.asList(
+                BtcECKey.fromPrivate(Hex.decode("fa01")),
+                BtcECKey.fromPrivate(Hex.decode("fa02"))
+        );
+        federation1Keys.sort(BtcECKey.PUBKEY_COMPARATOR);
+
+        Federation activeFederation = new Federation(
+                FederationTestUtils.getFederationMembersWithBtcKeys(federation1Keys),
+                Instant.ofEpochMilli(1000L),
+                0L,
+                btcRegTestParams
+        );
+
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+        BridgeSupport bridgeSupport = new BridgeSupport(
```

Comment:
You Should Use The BridgeSupportBuilder Class Whenever You Need An Instance Of The BridgeSupport For Testing

Reply:
@kelvinator07 These are pre-existing tests that were moved to this new class. 

I think we should always use BridgeSupportBuilder in new tests, for the existing ones it's a loooot of work to change them all. What I usually do is just change a couple of the existing ones when adding new tests. Gradually we'll get there one day